src/domain/UBGraphicsScene.cpp

```diff
@@ -3122,3 +3146,49 @@ void UBGraphicsScene::initStroke()
 {
     mCurrentStroke = new UBGraphicsStroke(shared_from_this());
 }
+
+void UBGraphicsScene::MultiTouchDrawing(QTouchEvent* event, UBStylusTool::Enum currentTool)
+{
+    QList <QTouchEvent::TouchPoint> touchPoints = event->touchPoints();
+    foreach (QTouchEvent::TouchPoint point, touchPoints)
+    {
+        lastPoint_m = point.lastPos();
+        endPoint_m = point.pos();
+
+        int distance = sqrt(pow((lastPoint_m.x() - endPoint_m.x()),2) + pow((lastPoint_m.y() - endPoint_m.y()),2)) + 1;
+        distance = sqrt(distance);
+        if (distance > 6)
+            distance = 6;
+        else if(distance < 4)
+            distance = 4;
+
+        UBBoardView* boardView = controlView();
+        QLineF line;
+        line.setP1(boardView->mapToScene(UBGeometryUtils::pointConstrainedInRect(lastPoint_m.toPoint(), boardView->rect())));
+        line.setP2(boardView->mapToScene(UBGeometryUtils::pointConstrainedInRect(endPoint_m.toPoint(), boardView->rect())));
+        if (!multiDrawLines.contains(line)) // to eliminate duplicates
+        {
+            multiDrawLines.append(line);
```

Comment:
This is important because there are so many duplicates that the page can take a long time to load

Reply:
If we are spending so much time on searching rather than appending, it might be worth using [`QSet`](https://doc.qt.io/qt-6/qset.html) instead of `QList`. `QSet` requires a bit more memory, but look-up is really fast.
Can you test it with `QSet` and report back whether it makes an appreciable difference in performance? Otherwise I'd just leave it as is.